lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure/State/Payment/Inspect.hs

```diff
@@ -58,17 +64,35 @@ import Prelude
 import qualified Data.Map.Strict as Map
 import qualified Data.Set as Set
 
+-- | Inspect the inputs and outputs of a transaction.
 data InspectTx = InspectTx
     { ourInputs :: [(TxId, TxIx, Coin)]
+    -- ^ Our inputs.
     , otherInputs :: [(TxId, TxIx)]
+    -- ^ Other inputs, there shouldn't be any.
     , change :: [(Address, Coin)]
+    -- ^ Change outputs.
     , ourOutputs :: [(Address, Customer, Coin)]
+    -- ^ Our outputs. The customer is the owner of the address. There could be
+    -- reasons the user wants to move funds among customer addresses.
     , otherOutputs :: [(Address, Coin)]
+    -- ^ Other outputs. This is regular money leaving the wallet.
     , fee :: Coin
     }
+    deriving (Eq, Show)
+
+-- | Calculate the output balance of a transaction, which is the sum of the
+-- values of the outputs minus the sum of the values of the change outputs.
+-- Eventual payments towards our addresses will be included in the output.
+-- Fee will be added to the inputs.
+transactionBalance :: InspectTx -> Coin
+transactionBalance InspectTx{..} =
```

Comment:
Might be interesting to express and check the `Property`s of the `transactionBalance` ?